wrappers/services.go

```diff
@@ -245,156 +185,141 @@ func StartServices(apps []*snap.AppInfo, disabledSvcs []string, flags *StartServ
-					err = sysd.Start([]string{socketService})
-				})
-			case snap.UserDaemon:
-				timings.Run(tm, "start-user-socket-service", fmt.Sprintf("start user socket service %q", socketService), func(nested timings.Measurer) {
-					err = startUserServices(cli, inter, socketService)
-				})
-			}
-			if err != nil {
-				return err
-			}
 		}
-
-		if app.Timer != nil {
-			timerService := filepath.Base(app.Timer.File())
-			// enable the timer
-			if err = sysd.Enable([]string{timerService}); err != nil {
-				return err
-			}
-
-			switch app.DaemonScope {
-			case snap.SystemDaemon:
-				timings.Run(tm, "start-system-timer-service", fmt.Sprintf("start system timer service %q", timerService), func(nested timings.Measurer) {
-					err = sysd.Start([]string{timerService})
-				})
-			case snap.UserDaemon:
-				timings.Run(tm, "start-user-timer-service", fmt.Sprintf("start user timer service %q", timerService), func(nested timings.Measurer) {
-					err = startUserServices(cli, inter, timerService)
-				})
-			}
-			if err != nil {
-				return err
-			}
+		if len(toEnableUser) > 0 {
+			err = userSysd.Enable(toEnableUser)
 		}
+	})
+	if err != nil {
+		return err
 	}
 
-	for _, srv := range systemServices {
-		// starting all services at once does not create a single
-		// transaction, but instead spawns multiple jobs, make sure the
-		// services started in the original order by bring them up one
-		// by one, see:
-		// https://github.com/systemd/systemd/issues/8102
-		// https://lists.freedesktop.org/archives/systemd-devel/2018-January/040152.html
-		timings.Run(tm, "start-service", fmt.Sprintf("start service %q", srv), func(nested timings.Measurer) {
-			err = systemSysd.Start([]string{srv})
-		})
-		if err != nil {
-			// cleanup was set up by iterating over apps
-			return err
+	timings.Run(tm, "start-services", "start services", func(nested timings.Measurer) {
+		for _, srv := range systemServices {
+			// starting all services at once does not create a
+			// single transaction, but instead spawns multiple jobs,
+			// make sure the services started in the original order
+			// by bringing them up one by one, see:
+			// https://github.com/systemd/systemd/issues/8102
+			// https://lists.freedesktop.org/archives/systemd-devel/2018-January/040152.html
+			timings.Run(tm, "start-service", fmt.Sprintf("start service %q", srv), func(nested timings.Measurer) {
+				err = systemSysd.Start([]string{srv})
+			})
+			if err != nil {
+				return
+			}
 		}
+	})
+	// let the cleanup know some services may have been started
+	servicesStarted = true
+	if err != nil {
+		// cleanup is handled in a defer
+		return err
 	}
 
 	if len(userServices) != 0 {
 		timings.Run(tm, "start-user-services", "start user services", func(nested timings.Measurer) {
 			err = startUserServices(cli, inter, userServices...)
 		})
+		// let the cleanup know some services may have been started
+		servicesStarted = true
```

Comment:
this was already set to `true` before

Reply:
actually the code above was confusing, I tweaked it a bit